### ISSUE TITLE
ArangoSearch date query returns wrong results if BM25 scoring function and other search fields are specified

### ISSUE THREAD

**Issue by andrea-tesei** [arangodb/arangodb#13149]
## My Environment

* __ArangoDB Version__: 3.7.3       
* __Storage Engine__: RocksDB            
* __Deployment Mode__: Single Server      
* __Deployment Strategy__: Manual Start in Docker  
* __Configuration__:               
* __Infrastructure__:  AWS             
* __Operating System__:  Amazon Linux 2 (Red Hat based)      
* __Total RAM in your machine__:  8GB      
* __Disks in use__:  General Purpose SSD      
* __Used Package__:  Docker - official Docker library            

## Component, Query & Data

__Affected feature__:
ArangoSearch search date field query **ONLY** with other non-date query field and scoring function enabled (BM25)

__AQL query (if applicable)__:
```
FOR doc in doc_view SEARCH ( IN_RANGE( doc.date, "2013-12-01","2013-12-31", true, true) AND  doc.field_2=="value_1") 
LET score = BM25(doc) 
SORT doc.date DESC
RETURN {score: score, date: doc.date}
```
__AQL explain and/or profile (if applicable)__:

__Dataset__:

```
The documents are composed by two keys:
db.collection.save({
date: "DATE_STRING", // Format: YYYY-MM-dd
field_1: "string_value", // The type of this field is not relevant for the query
})
```
__Size of your Dataset on disk__:

1.2 GB

Two collections are specified in view links:
* The affected collection (i.e. the one that contains 'data' field) has 115K documents
* The other collection has 14MLN documents

__Replication Factor & Number of Shards (Cluster only)__:

## Steps to reproduce

1. Index a set of documents with date and other fields
2. Create persistent indexes in the date and other fields
3. Execute query with BM25 scoring function active

__Problem__:
When the ArangoSearch query containing IN_RANGE function is executed in conjunction with other field query and BM25 scoring function, the result set contains also documents with date value outside specified range. 

I've tried to disable the scoring function: in this case the result set contains the **correct** results. Even if I remove the field_2 filter item in ArangoSearch query, thus executing only IN_RANGE function, the result set contains the **correct** results.

I've tried to replicate this behavior with a test dataset, but unfortunately with few documents the query results are always correct. I think this behavior arise only with a large number of indexed documents.

__Expected result__:
The result set should contains **ONLY** documents that matches the specified date range.

**Comment by Simran-B** [comment]
@andrea-tesei Can you check if it's reproducible with the same dataset but in a fresh ArangoDB instance? If yes, would it be possible for you to share the dataset with us (in private)?

**Comment by andrea-tesei** [comment]
Hi @Simran-B thanks for your reply. The problem is reproducible with a fresh ArangoDB instance: I've used arangorestore to setup the current instances affected by the problem. 

For what concerns sharing the dataset, I will follow-up this conversation on the email you gave me. Thanks

**Comment by andrea-tesei** [comment]
Hi guys. Happy New Year!

I've successfully uploaded the dataset in private. Have you had time to reproduce the issue and investigate it? I look forward to hearing from you. Thanks

**Comment by maxkernbach** [comment]
Hi @andrea-tesei,

Happy New Year and thank you for providing the dataset. Following your instructions, I was able to reproduce the issue as described with 3.7.3. If `BM25()` is used, documents with an invalid date are returned.

However, the latest version 3.7.6, which includes various ArangoSearch fixes, does not show this behaviour. The same (correct) document set is returned, also when using the `BM25()` function.

Could you please upgrade to 3.7.6 and let us know if this already fixes the issue?

**Comment by Simran-B** [comment]
@andrea-tesei Can you verify that it's fixed with 3.7.6? Thanks!

**Comment by andrea-tesei** [comment]
Hi @Simran-B and @maxkernbach. Yes we have done some basic tests and I confirm that the issue seems to be fixed in 3.7.6. I was waiting the end of further test sessions in production to give you the final feedback. Anyway, you can close this issue for now. Thanks for your effort and keep the dataset for further tests if it can help in the future.

**Comment by maxkernbach** [comment]
Hi @andrea-tesei,

Thanks for your feedback, I am closing this issue for now.
